### PR TITLE
DevDocs: Modernize and clean devdocs serverside route

### PR DESCRIPTION
**Pull request by withinboredom**
While working on some devdocs serverside things in a [separate PR](https://github.com/Automattic/wp-calypso/pull/9530) I found this file littered with ESLint errors. This PR fixes this errors and modernizes the code a bit.

Each commit is independent, so if switching to arrow functions (or something else) is too much, that singular commit can be reverted.

cc: @ehg

### CODE REVIEW

**Comment by withinboredom** [comment]
Test Live: https://calypso.live/?branch=fix/modernize-and-clean-devdocs-route

File: server/devdocs/index.js [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-	lunr = require( 'lunr' ),
-	find = require( 'lodash/find' ),
-	escapeHTML = require( 'lodash/escape' );
+
+import express from 'express';
+import fs from 'fs';
+import fspath from 'path';
+import marked from 'marked';
+import lunr from 'lunr';
+import find from 'lodash/find';
```
Comment (gwwar) [inline server/devdocs/index.js]: In Calypso we use the`babel-plugin-lodash` transform that does this for us, so we can also write:
```javascript
import { find, escape as escapeHTML } from 'lodash';
```

Comment (withinboredom) [inline server/devdocs/index.js]: Thanks for that, fixing it up now

**Comment by gwwar** [comment]
👍 This looks good in a local docker env and nothing exploded in a local Desktop env.

**Pull request merged**